matroid/serving
# Automatic Update of models being served

inception_inference_dynamic creates a service which dynamically scans directory to look for updated models. 
inception_inference.proto changed to allow for passing of network_id to run inference for a single network. This needs to be integrated from client side (matroid repo) to serve the right path and call classify with network_id

To test from terminal,
cd ~/serving
git pull origin
git checkout add_dynamic_inception_serving
git submodule update

##Serving Script
#Build
cd ~/serving/
bazel build  -c opt  //tensorflow_serving/inception_example:inception_inference_dynamic
#Run
~/serving/bazel-bin/tensorflow_serving/inception_example/inception_inference_dynamic --port=9031 ~/matroid/scheduler/model_serving/<model-id>/

##Client Script
#Build
cd ~/serving/
bazel build  -c opt  //tensorflow_serving/inception_example:inception_client
#Run
~/serving/bazel-bin/tensorflow_serving/inception_example/inception_client --server=localhost:9031 --image_file=/tmp/imagenet/cropped_panda.jpg:/tmp/imagenet/cropped_panda.jpg

Test Dynamic Serving:
#Copy new model to path in version 2
cp -r ~/matroid/scheduler/model_serving/<model-id-2>/00000001/ ~/matroid/scheduler/model_serving/<model-id>/00000002/ 
#Run the client script with same image
~/serving/bazel-bin/tensorflow_serving/inception_example/inception_client --server=localhost:9031 --image_file=/tmp/imagenet/cropped_panda.jpg:/tmp/imagenet/cropped_panda.jpg

You should get different results for the 2nd inference run as you are using a different model
